### DISCUSSION TITLE
Displaying a "placeholder" page when resource is down (502 code)

### DISCUSSION THREAD

**Discussion by mrfuntime** [NginxProxyManager/nginx-proxy-manager#1797] [Q&A] [answered]
There are times when I have a resource(s) behind NPM down legitimately for whatever reason.  I'd like NPM to serve up a nice looking "System currently unavailable" page instead of the standard 502 Bad Gateway page.

Has anyone done this?  Can it be done??  I'm not sure where to start.

Thanks in advance!!
Mike

**Comment by flod1** [comment]
exactly what I would like to implement

**Comment by GerritKuilder** [comment, marked as answer]
Hi,
in my `nginx/custom` directory I have a server_proxy.conf file

in that file I have:

```
error_page 400 /error_pages/400.html;
error_page 401 /error_pages/402.html;
error_page 402 /error_pages/402.html;
error_page 403 /error_pages/403.html;
error_page 404 /error_pages/404.html;
error_page 500 /error_pages/500.html;
error_page 501 /error_pages/501.html;
error_page 502 /error_pages/502.html;
error_page 503 /error_pages/503.html;
proxy_intercept_errors on;

location /error_pages/ {
    alias /data/nginx/error_pages/;
    internal;
}
```
the actual path to the error pages is: `nginx/error_pages`

I got the error pages from github somewhere.

HTH, Gerrit

**Reply by BobWs** [reply]
@GerritKuilder So your solution is working globally for all the proxy hosts?
As I understand your description you need to create to folders? One folder called “custom” and one called “error_pages” so in the custom folder you place the server_proxy.conf file with the code above and in the error_pages folder you place all the custom error html files. Is this the correct way to do it?

**Comment by BobWs** [comment]
I have created the custom folder in/data/nginx and placed the server_proxy.conf file within, but when I restart the container I get an error message:
`nginx: [emerg] "proxy_intercept_errors" directive is duplicate in /data/nginx/custom/server_proxy.conf:10 `

This is how I mapped the volumes to the host:
````
    volumes:
      - /volume1/docker-sync/npm/internal/config/config.json:/app/config/production.json
      - /volume1/docker-sync/npm/internal/data:/data
      - /volume1/docker-sync/npm/internal/letsencrypt:/etc/letsencrypt
      - /etc/localtime:/etc/localtime:ro
      - /etc/TZ:/etc/timezone:ro

````
How do I solve this?
TIA

**Reply by GerritKuilder** [reply]
this is my volumes config:
````
 volumes:
      - ./data:/data
      - ./letsencrypt:/etc/letsencrypt
      - ./modules:/etc/nginx/modules

````

**Reply by GerritKuilder** [reply]
Path to error pages: [proxy dir]data/nginx/error_pages
I don't have anything specific for 'config/config.js' So maybe you have to look at that file.
